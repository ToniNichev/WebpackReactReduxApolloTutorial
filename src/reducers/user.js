const CHANGE_USERNAME = 'CHANGE_USERNAME';
const TOGGLE_EDIT_MODE = 'TOGGLE_EDIT_MODE';

const initialState = {
  todos: [],
  userName: "No Name",
  editMode: false
};

  const reducer = (state = initialState, action) => {
    switch (action.type) {
      case CHANGE_USERNAME: {
        let newState = {...state};
        newState.userName = action.data;
        return newState;
      }

      case TOGGLE_EDIT_MODE: {
        let newState = {...state};
        newState.editMode = !newState.editMode;
        return newState; 
      }

      default:
        return state;
    }
  };

export default reducer;