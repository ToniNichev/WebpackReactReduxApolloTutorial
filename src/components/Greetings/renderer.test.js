import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import reducers from '../../reducers';
import { createStore} from 'redux';
import Component from './renderer';

const store = createStore(reducers, {});
let wrapper;
describe('Testing Greetings component', () => {

<<<<<<< HEAD
  beforeEach(() => {
    wrapper = mount(
        <Component styles={{}} store={store} />
=======
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Component styles={{}} />
      </Provider>
>>>>>>> 6e17888c591d28acac720d168a60602f471d57d6
    );
  });

  it('renders as expected', () => {
    // to snapshot test
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('textbox does not exist', () => {  
    expect(wrapper.find('input').exists()).toBe(false);
  });

  it('textbox exists after h2 click', () => {  
    // simulate click on the h2 tag trigering props change and visualizing the input text box
    console.log(">>>"+ wrapper.find('h2').html() + "<<<");
    wrapper.find('h2').simulate('click');
    console.log(">>>"+ wrapper.find('h2').html() + "<<<");
    expect(wrapper.find('input').exists()).toBe(true);
  });    

  it('textbox values change properly', () => {    
    // text box value tests
    expect(wrapper.find('input').props().value).toBe('No Name');
    wrapper.find('input').props().value = 'test';
    expect(wrapper.find('input').props().value).toBe('test');
  });    

});
