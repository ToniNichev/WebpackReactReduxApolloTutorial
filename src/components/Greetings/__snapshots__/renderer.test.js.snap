// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Greetings component renders as expected 1`] = `
<Connect(Greetings)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  styles={Object {}}
>
  <Greetings
    changeUserName={[Function]}
    editMode={false}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    styles={Object {}}
    toggleLogInPopup={[Function]}
    userName="No Name"
  >
    <div>
      <div>
        <h2
          onClick={[Function]}
        >
          Hello:  
          No Name
        </h2>
      </div>
    </div>
  </Greetings>
</Connect(Greetings)>
`;
